### PR TITLE
Core breaking changes improvements

### LINKED ISSUE #15994: Clean up .NET Core app compat section
- rename to "breaking changes"
- add a 2.2-3.1 upgrade path file
- remove preview versions from selectors
- remove preview upgrade path files
- move WinForms topic to porting section of TOC

### PR DESCRIPTION
**Pull request by gewarren**
- Rename "Application compatibility" in TOC to "Breaking changes" in Core TOC
- Remove unused top-level node from breaking changes TOC
- Sentence casing in Core TOC
- Add 2.2 to 3.1 migration topic (containing all includes from 2.2-3.0 and 3.0-3.1)
- Change "category" to "area" since it was overloaded
- Order invidividual changes from most recent to least recent within a topic (previously vice versa)
- Delete preview version breaking changes topics
- Remove preview versions from version selector
- Move breaking changes includes into version subfolders (to match what ASP.NET Core does)
- Delete extraneous "crypto.md" file

Fixes #15994

[Preview link](https://review.docs.microsoft.com/en-us/dotnet/core/compatibility/breaking-changes?branch=pr-en-us-15996)

### CODE REVIEW

File: docs/core/compatibility/toc.yml [hunk 1]
```diff
@@ -1,42 +1,32 @@
-- name: Application compatibility
+- name: Evaluate breaking changes
+  href: index.md
+- name: Categories
+  href: categories.md
+- name: Page selectors
```
Comment (tdykstra) [inline docs/core/compatibility/toc.yml]: This TOC node "page selectors" goes to an H1 ".NET Core breaking changes" which is a jarring disconnect and the H1 doesn't really match the page content. Maybe make the H1 something like "Page selectors for .NET Core breaking changes"?

That said, the page duplicates navigation provided by the TOC, and SkyRay shows less than 400 page views for it - I would consider deleting it.

**Review by tdykstra — APPROVED** [review]
LGTM, a couple of suggestions.

File: docs/core/compatibility/toc.yml (hunk 1 above)
Comment (gewarren) [reply docs/core/compatibility/toc.yml]: I changed the H1 to "Breaking change selectors".

File: docs/core/compatibility/toc.yml [hunk 2]
```diff
@@ -1,42 +1,32 @@
-- name: Application compatibility
+- name: Evaluate breaking changes
```
[the remark this answers is not in the document]
Comment (gewarren) [reply docs/core/compatibility/toc.yml]: I changed the TOC name to "What are breaking changes?"

**Comment by gewarren** [comment]
Thanks for the review @tdykstra!

**Pull request merged**